From 316cabd72c7341a2475684627e2f9918e2659996 Mon Sep 17 00:00:00 2001
From: AlexZorzi <alleszorzi@gmail.com>
Date: Sat, 19 Sep 2026 00:58:50 +0200
Subject: [PATCH] Load obfuscated resources sanely

Fall back to ResourcesCompat.getDrawable when the obfu_res/ asset is
missing, instead of letting FileNotFoundException escape.

Ported from myron's 0002; x8/a -> Se/a, and ResourcesCompat is R8-renamed
to W/f$a here.
---
 smali_classes2/Se/a.smali | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/smali_classes2/Se/a.smali b/smali_classes2/Se/a.smali
index 50d2dc0..d17f595 100644
--- a/smali_classes2/Se/a.smali
+++ b/smali_classes2/Se/a.smali
@@ -163,7 +163,7 @@
 .end method
 
 .method public static c(Landroid/content/res/Resources;ILjava/lang/Integer;)Landroid/graphics/drawable/Drawable;
-    .locals 5
+    .locals 6
 
     invoke-virtual {p0, p1}, Landroid/content/res/Resources;->getResourceTypeName(I)Ljava/lang/String;
 
@@ -225,6 +225,8 @@
     move p2, v2
 
     :goto_0
+    move v5, p1
+
     const/4 v3, 0x1
 
     invoke-virtual {p0, p1, p2, v0, v3}, Landroid/content/res/Resources;->getValueForDensity(IILandroid/util/TypedValue;Z)V
@@ -245,9 +247,13 @@
 
     move-result-object p2
 
+    :try_start_9
+
     invoke-virtual {p2, p1}, Landroid/content/res/AssetManager;->open(Ljava/lang/String;)Ljava/io/InputStream;
 
     move-result-object p2
+    :try_end_9
+    .catch Ljava/io/FileNotFoundException; {:try_start_9 .. :try_end_9} :catch_9
 
     :try_start_0
     invoke-static {p2}, Lkotlin/jvm/internal/l;->d(Ljava/lang/Object;)V
@@ -304,6 +310,17 @@
 
     goto :goto_3
 
+    :catch_9
+    move-exception p2
+
+    const/4 p2, 0x0
+
+    invoke-static {p0, v5, p2}, LW/f$a;->a(Landroid/content/res/Resources;ILandroid/content/res/Resources$Theme;)Landroid/graphics/drawable/Drawable;
+
+    move-result-object p0
+
+    return-object p0
+
     :goto_2
     :try_start_3
     throw p0
